File: backend/app/schemas/student.py
from pydantic import BaseModel
from typing import Optional


class StudentProgressResponse(BaseModel):
    total_answered: int
    correct_count: int
    accuracy: Optional[float] = None
    standards_attempted: int


class DailyGoalResponse(BaseModel):
    target: int
    answered_today: int
    correct_today: int
    remaining: int
    completed: bool
    progress: float
    message: str


class SkillMapDomainResponse(BaseModel):
    domain_id: int
    domain_name: str
    domain_code: str
    progress: float
    level: str
    level_description: str
    questions_attempted: int
    correct_count: int
    incorrect_count: int
    accuracy: Optional[float] = None
    correct_streak: int
    total_standards: int
    active_questions: int
    recommended: bool
    recommendation_reason: str
    sort_priority: int

File: frontend/src/types/student.ts
export interface StudentProgress {
  total_answered: number
  correct_count: number
  accuracy: number | null
  standards_attempted: number
}

export interface AnswerSubmit {
  question_id: number
  selected_answer: string
  is_correct: boolean
}

export interface DailyGoal {
  target: number
  answered_today: number
  correct_today: number
  remaining: number
  completed: boolean
  progress: number
  message: string
}

export interface SkillMapDomain {
  domain_id: number
  domain_name: string
  domain_code: string
  progress: number
  level: string
  level_description: string
  questions_attempted: number
  correct_count: number
  incorrect_count: number
  accuracy: number | null
  correct_streak: number
  total_standards: number
  active_questions: number
  recommended: boolean
  recommendation_reason: string
  sort_priority: number
}
